urld0f681c558_farom57_AstroidDriver_tgz-pJ8-laazotea_indi_server_examples_MinimumINDIServerJ8/src/laazotea/indi/driver/examples/INDIElTiempoDriver.java

```diff
@@ -0,0 +1,451 @@
+
+    int pos1 = fileName.lastIndexOf("-");
+
+    String name = fileName.substring(pos1, pos1 + 5) + "/" + fileName.substring(pos1 + 5, pos1 + 7) + "/" + fileName.substring(pos1 + 7, pos1 + 9) + " " + fileName.substring(pos1 + 9, pos1 + 11) + ":" + fileName.substring(pos1 + 11, pos1 + 13);
+
+    INDITextProperty pn = (INDITextProperty) getProperty(imagePrefix + "_IMAGE_NAME");
+    INDITextElement en = (INDITextElement) pn.getElement(imagePrefix + "_IMAGE_NAME");
+
+    en.setValue(imagePrefix + " Satellite " + name + " UTC");
+
+    return true;
+  }
+
+  /**
+   * Reads a text file and returns its contents as a String.
+   *
+   * @param file The file to read
+   * @return The contents of the file
+   * @throws IOException
+   */
+  private String readFile(File file) throws IOException {
+    BufferedReader reader = new BufferedReader(new FileReader(file));
+    String line;
+    StringBuilder stringBuilder = new StringBuilder();
+    String ls = System.getProperty("line.separator");
+
+    while ((line = reader.readLine()) != null) {
+      stringBuilder.append(line);
+      stringBuilder.append(ls);
+    }
+
+    return stringBuilder.toString();
+  }
+
+  /**
+   * Reads a binary file and returns its contents as a byte[]
+   *
+   * @param file The file to be read
+   * @return The contents of the file
+   * @throws IOException
+   */
+  private byte[] readBinaryFile(File file) throws IOException {
+    int fileSize = (int) file.length();
+    FileInputStream reader = new FileInputStream(file);
+
+    byte[] buffer = new byte[fileSize];
+
+    int totalRead = 0;
+
+    while (totalRead < fileSize) {
+      int readed = reader.read(buffer, totalRead, fileSize - totalRead);
+
+      if (readed == -1) {
+        return null; // Unexpected end of file 
+      }
+
+      totalRead += readed;
+    }
+
+    return buffer;
+  }
+
+  /**
+   * Downloads a
+   * <code>url</code> and saves its contents to
+   * <code>file</code>.
+   *
+   * @param url
+   * @param file
+   * @throws IOException
+   * @throws MalformedURLException
+   */
+  private void downloadAndSave(String url, File file) throws IOException, MalformedURLException {
+    int bufsize = 65536;
+    byte[] buffer = new byte[bufsize];
+
+    URL u = new URL(url);
+    InputStream is = u.openStream();  // throws an IOException
+    BufferedInputStream bis = new BufferedInputStream(is);
+    FileOutputStream fos = new FileOutputStream(file);
+
+    int readed = 0;
+
+    while (readed != -1) {
+      readed = bis.read(buffer);
+
+      if (readed > 0) {
+        fos.write(buffer, 0, readed);
+      }
+    }
+
+    bis.close();
+    fos.close();
+  }
+
+  /**
+   * The thread that every 15 minutes checks for new images and, if they have
+   * changed, downloads them and sends them back to the clients.
+   */
+  @Override
+  public void run() {
+    while (!stop) {
+      try {
+        Thread.sleep(15 * 60 * 1000);
+      } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)